pyannote/audio/labeling/tasks/multilabel.py

```diff
@@ -0,0 +1,306 @@
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# AUTHORS
+# Marvin Lavechin - marvinlavechin@gmail.com
+
+"""BabyTrain
+4-way classification :
+KCHI    (the key child wearing the device)
+CHI     (other children)
+FEM     (female speech)
+MAL     (male speech)
+"""
+
+from itertools import cycle
+
+import numpy as np
+import torch
+from pyannote.audio.features import Precomputed
+from pyannote.core import Annotation
+from pyannote.core import SlidingWindowFeature
+from pyannote.core import Timeline
+from pyannote.core.utils.numpy import one_hot_encoding
+from pyannote.database import get_annotated
+from pyannote.database import get_protocol
+
+from .base import LabelingTask
+from .base import LabelingTaskGenerator
+from .. import TASK_MULTI_LABEL_CLASSIFICATION
+
+
+def derives_label(annotation, derivation_type, meta_label, regular_labels):
+    """
+    Derives a label. The derivation takes as inputs :
+    - An annotation from which we want to derive
+    - A derivation type : union or intersection
+    - A meta label : the name of the output label
+    - A list of regular labels : the regular labels from which we want to derive
+
+    Example :
+        1) derives_label(annotation, 'union', 'speech', ["CHI","MAL","FEM"]
+        Will compute the speech label based on the union of "CHI", "MAL" and "FEM"
+        2) derives_label(annotation, 'intersection', 'overlap', ["CHI","MAL","FEM"]
+        Will compute the overlapping speech based on the intersection of "CHI", "MAL" and "FEM"
+
+    annotation:  Annotation type
+        The annotation we want to derive from.
+    derivation_type: string, must belong to ['union', 'intersection']
+        The derivation type
+    meta_label:  string
+        The meta label, the name of the label returned by the derivation
+    regular_labels:  list of strings
+        A list of regular labels we want to derive from
+    """
+    if derivation_type not in ['union', 'intersection']:
+        raise ValueError("Derivation type must be in ['union', 'intersection')")
+
+    derived = Annotation()
```

Comment:
Whenever possible try to keep the "uri" unchanged.
```suggestion
    derived = Annotation(uri=annotation.uri)
```